src/platform_impl/macos/mod.rs

```diff
@@ -1 +1,526 @@
+mod accelerator;
 
+use std::{cell::RefCell, collections::HashMap, rc::Rc};
+
+use cocoa::{
+    appkit::{NSApp, NSApplication, NSEventModifierFlags, NSMenu, NSMenuItem},
+    base::{id, nil, selector, NO, YES},
+    foundation::{NSAutoreleasePool, NSInteger, NSString},
+};
+use objc::runtime::{Object, Sel};
+
+use crate::{
+    accelerator::Accelerator,
+    MenuEntryType,
+    predefined::PredfinedMenuItemType,
+    util::{AddOp, Counter},
+};
+
+static COUNTER: Counter = Counter::new();
+
+/// A generic child in a menu
+///
+/// Be careful when cloning this item and treat it as read-only
+#[derive(Debug, Default)]
+#[allow(dead_code)]
+struct MenuChild {
+    // shared fields between submenus and menu items
+    type_: MenuEntryType,
+    text: String,
+    enabled: bool,
+
+    ns_menuitems: HashMap<Menu, id>,
```

Comment:
Yes, adding the same item to multiple menus causes exceptions unfortunately.

One thing that's not solved by the current implementation is adding the same item *to the same menu* multiple times. Is that something we care about…? What did you do about that on GTK? (Guess I can just have a look…)

Reply:
I should also note that I'm going to have to change `HashMap<Menu, id>` to `HashMap<u32, id>` and index by the parent menu's `id` field instead, because I need to add more fields to `Menu` and it will become hard to derive `Eq` which is needed for it to be a `HashMap` key.

Reply:
On gtk, each item has a hashmap whefe each key is the parent menu id and the value is a vector of gtk items, in macOS case, it will be ns_menuitem